### DISCUSSION TITLE
ToggleButtons ->multiple() validation max selected items?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#12442] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.71

### How can we help you?

`->rules(['max:3'])` does not work

```
ToggleButtons::make('technologies')
    ->multiple()
    ->options([
        'tailwind' => 'Tailwind CSS',
        'alpine' => 'Alpine.js',
        'laravel' => 'Laravel',
        'livewire' => 'Laravel Livewire',
    ])
```

**Comment by dissto** [comment, marked as answer]
I cannot reproduce your error:

![Screenshot 2024-04-22 151629](https://github.com/filamentphp/filament/assets/11778632/7f1dd684-3f92-4d6e-8cfd-571047aa04ec)

```php
ToggleButtons::make('technologies')
  ->multiple()
  ->options([
      'tailwind' => 'Tailwind CSS',
      'alpine' => 'Alpine.js',
      'laravel' => 'Laravel',
      'livewire' => 'Laravel Livewire',
  ])
  ->rules('max:3'),
```
The validation message is maybe misleading, but that could easily be dealt with. 🤔

**Reply by micraux** [reply]
You can have a [custom validation massage](https://filamentphp.com/docs/3.x/forms/validation#validation-messages)
